src/realmd/AuthSocket.cpp

```diff
@@ -539,7 +547,8 @@ bool AuthSocket::_HandleLogonProof()
     DEBUG_LOG("Entering _HandleLogonProof");
     ///- Read the packet
     std::shared_ptr<sAuthLogonProof_C> lp = std::make_shared<sAuthLogonProof_C>();
-    Read((char*)lp.get(), sizeof(sAuthLogonProof_C), [self = shared_from_this(), lp](const boost::system::error_code& error, std::size_t read)
+    auto size = m_promptPin ? sizeof(sAuthLogonProof_C) : sizeof(sAuthLogonProof_C) - sizeof(PINData);
```

Comment:
Cyberium had a good point that these sizeof magics should be a function of the struct.

Reply:
I honestly don't know how I would accomplish that

Reply:
size_t size = lp->getsize(bool withPin)